File: typescript/src/renderer.ts
import { Shape } from './shapes/shape.js'

function newRenderer(shapeToRender: Shape) {
  return {
    draw() {
      const area: number = shapeToRender.computeArea()
      console.log('Shape drawn\n' + 'Its area is ' + area)
    }
  }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    public Shape targetShape;
    
    Renderer(Shape targetShape) {
        this.targetShape = targetShape;
    }

    void draw() {
        double area = targetShape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
